### PR TITLE
Fill in sqlExpression for functions related SQL entries

### PR DESCRIPTION
**Pull request by jakipatryk**

### CODE REVIEW

File: ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionBody.scala [hunk 1]
```diff
@@ -15,10 +15,45 @@
...
+       |  RETURNS $returnType
+       |AS
+       |$$$$
+       |BEGIN
+       |${bodyCode.value}
+       |END;
+       |$$$$
+       |LANGUAGE plpgsql VOLATILE SECURITY DEFINER;
+       |""".stripMargin
+  }
```
Comment (dk1844) [inline ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionBody.scala:54]: I wonder if we need to recreate the function SQL at all. Why don't we just save it whole and pass it as-is?

**Review by dk1844 — APPROVED** [review]
LGTM

File: ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionBody.scala (hunk 1 above)
Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionBody.scala:54]: Yeah maybe. But then we don't need drops, getting current function metadata from PG. Hmm

Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionBody.scala:54]: I mean if we just do `CREATE OR REPLACE FUNCTION` in SQL file.

**Review by Zejnilovic — APPROVED** [review]

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala [hunk 2]
```diff
@@ -15,18 +15,19 @@
  */
 package za.co.absa.ultet.dbitems
 
-import za.co.absa.ultet.model.SQLEntry
+import za.co.absa.ultet.model
+import za.co.absa.ultet.model.{SQLEntry, function}
 
 case class DBFunctionFromPG(sqlEntries: Seq[SQLEntry]) extends DBFunction {
-  override def fnName: String = ???
+  override def fnName: function.FunctionName = ???
```
Comment (benedeki) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala:22]: Why keep the prefixes?

File: ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionGrant.scala [hunk 3]
```diff
@@ -15,17 +15,25 @@
...
+       |  ${schemaName.value}.${functionName.value}($argumentsString)
+       |TO
+       |  ${userToGrantExecuteTo.value};
+       |""".stripMargin
+  }
 
   override def transactionGroup: TransactionGroup = ???
 
-  override def orderInTransaction: Int = ???
+  override def orderInTransaction: Int = 102
```
Comment (benedeki) [inline ultet/src/main/scala/za/co/absa/ultet/model/function/FunctionGrant.scala:38]: **Minor:**
Would swap the order of grants and ownership

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala (hunk 2 above)
Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala]: IntelliJ, fixed

**Pull request merged**